### ISSUE TITLE
`@` attribute is not at the highest level

### ISSUE THREAD

**Issue by yegor256** [objectionary/lints#108]
Consider this code:

```
[] > foo
  hello > x
    bar > @
```

It compiles, but it's very confusing. It's better to rewrite it like this:

```
[] > foo
  bar > @
  hello > x
    @
```

Let's try to create a lint for this.

**Comment by Marat-Tim** [comment]
@yegor256 Do we really need this lint?
We already have `phi-is-not-first`([issue](https://github.com/objectionary/lints/issues/109)), and if it's working correctly, it should prevent this kind of construction anyway

**Comment by yegor256** [comment]
@Marat-Tim how would the error message sound, from `phi-is-not-first`, for this code?

```
[] > foo
  hello > x
    bar > @
```

**Comment by Marat-Tim** [comment]
@yegor256 `[foo phi-is-not-first (Single) WARNING]:4 The "@" object must go first (except void attributes ordering)`

**Comment by Marat-Tim** [comment]
@yegor256 hmm, I thought about it - turns out `phi-is-not-first` doesn’t cover all cases
For example, there’s no error in code like this one
```eo
# Comment.
[] > foo
  hello
    bar > @
```

**Comment by yegor256** [comment]
@Marat-Tim as you see, this error message is not very helpful in this case. The user will be confused.

**Comment by Marat-Tim** [comment]
@yegor256 It looks like this is impossible to implement now
Take a look at this EO code and its XMIR output
```eo
# Comment.
[] > app
  foo > x
    bar > @
```
xmir
```xml
<o line="2" name="app" pos="0">
   <o base="Q.org.eolang.foo" line="3" name="x" pos="2">
      <o base="$.@" line="4"/>
   </o>
   <o base="Q.org.eolang.bar" line="4" name="@" pos="4"/>
</o>
```
and then compare it to this other EO code and its XMIR
```eo
# Comment.
[] > app
  foo > x
    @
  bar > @
```
xmir
```xml
<o line="2" name="app" pos="0">
   <o base="Q.org.eolang.foo" line="3" name="x" pos="2">
      <o base="$.@" line="4" pos="4"/>
   </o>
   <o base="Q.org.eolang.bar" line="5" name="@" pos="2"/>
</o>
```
The only difference is in the `line` and `pos` attributes
Can we rely on those for linting?
(we can check that the object named `@` has an indentation(`pos`) exactly 2 more than its parent)

**Comment by yegor256** [comment]
@Marat-Tim in this case, let's rely on `pos`, I agree

**Comment by Marat-Tim** [comment]
@yegor256 It turns out it's not that simple. If we don't just have an object, but are retrieving an attribute from another object, then `pos` will be the position of the last dot in the `base` declaration
```eo
# Comment.
[] > app
  bipki.x > @
```
xmir
```xml
<o line="2" name="app" pos="0">
   <o base="Q.org.eolang.bipki.x" line="3" name="@" pos="7"/>
</o>
```
(`pos` equal 7, not 2)

We can calculate the correct position by subtracting the length of the base attribute up to the last dot from the `pos` attribute
However, we will need to consider that the object might start with `Q.org.eolang`, `$`, or other special characters and ignore them when calculating

Is it worth it?

**Comment by Marat-Tim** [comment]
@yegor256 Actually no, we can't distinguish this case
```eo
# Comment.
[] > app
  bipki.x > @
  [] > bipki
    bar > x
```
xmir
```xml
<o line="2" name="app" pos="0">
   <o base="$.bipki.x" line="3" name="@" pos="7"/>
   <o line="4" name="bipki" pos="2">
      <o base="Q.org.eolang.bar" line="5" name="x" pos="4"/>
   </o>
</o>
```
from this one
(
```diff
-   bipki.x > @
+   $.bipki.x > @
```
)
```eo
# Comment.
[] > app
  $.bipki.x > @
  [] > bipki
    bar > x
```
xmir
```xml
<o line="2" name="app" pos="0">
   <o base="$.bipki.x" line="3" name="@" pos="9"/>
   <o line="4" name="bipki" pos="2">
      <o base="Q.org.eolang.bar" line="5" name="x" pos="4"/>
   </o>
</o>
```
I have no idea how to implement this lint

**Comment by VasilevNStas** [comment]
It turns out this case is already covered by the existing `names/phi-is-not-first.xsl` lint. Let me explain how.

The key insight: when `@` is declared nested (inside a child object), the parser floats it up to the parent level, but the floated `φ` object does **not** become the first sibling — its container always precedes it. And `phi-is-not-first` checks precisely that the `φ` object goes first (except void attributes ordering).

Bad code from this issue:
```
[] > foo
  hello > x
    bar > @
```
XMIR (after parsing):
```xml
<o line="1" name="foo" pos="0">
  <o base="Φ.hello" line="2" name="x" pos="2"/>
  <o base="Φ.bar" line="3" name="φ" pos="4"/>
</o>
```
The lint reports: `defect line="3"` — exactly the nested `@`.

The recommended rewrite:
```
[] > foo
  bar > @
  hello > x
    @
```
XMIR:
```xml
<o line="1" name="foo" pos="0">
  <o base="Φ.bar" line="2" name="φ" pos="2"/>
  <o base="Φ.hello" line="3" name="x" pos="2">
    <o as="α0" base="ξ.φ" line="4" pos="4"/>
  </o>
</o>
```
Now `φ` is the first sibling — no defect.

I verified on the real corpus (186 `.eo` files from `objectionary/eo`, `phino`, `jeo-maven-plugin`): the lint fires on 4 files only (2 intentionally-broken benchmarks `*-rust.eo` + `app.eo`, `simple-rust.eo`), with no false positives. The bare `@` inside a child object (`hello > x / @`) is a legitimate reference to the parent's φ, not a nested declaration, and correctly produces no defect.

So the suggested fix in this issue should already produce the expected warning; only a regression test-pack for the nested case is missing (existing packs cover the ordering case only).
